src/Microsoft.AspNetCore.Routing/InvalidRouteException.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Routing
+{
+    public class InvalidRouteException : Exception
```

Comment:
I'd rather be more conservative and write code for stuff that we know happens. It sounds like @rynowak has some thoughts on other scenarios we care about. But this type name doesn't seem like a good match.